parlai/core/torch_classifier_agent.py

```diff
@@ -0,0 +1,285 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+from parlai.core.torch_agent import TorchAgent, Output
+from parlai.core.utils import round_sigfigs, warn_once
+from collections import defaultdict
+
+import torch
+import torch.nn.functional as F
+
+
+class TorchClassifierAgent(TorchAgent):
+    """Abstract Classifier agent. Only meant to be extended.
+
+    TorchClassifierAgent aims to handle much of the bookkeeping any
+    classification model.
+    """
+
+    @staticmethod
+    def add_cmdline_args(parser):
+        TorchAgent.add_cmdline_args(parser)
+        parser = parser.add_argument_group('Torch Classifier Arguments')
+        # class arguments
+        parser.add_argument('--classes', type=str, nargs='*', default=None,
+                            help='the name of the classes.')
+        parser.add_argument('--class-weights', type=float, nargs='*', default=None,
+                            help='weight of each of the classes for the softmax')
+        parser.add_argument('--ref-class', type=str, default=None, hidden=True,
+                            help='the class that will be used to compute '
+                                 'precision and recall. By default the first '
+                                 'class.')
+        parser.add_argument('--threshold', type=float, default=0.5,
+                            help='during evaluation, threshold for choosing '
+                                 'ref class; only applies to binary '
+                                 'classification')
+        # interactive mode
+        parser.add_argument('--interactive-mode', type='bool', default=False)
```

Comment:
hmm, we can't infer this from whether a label is provided?

Reply:
i suppose, the reason i made it explicit is that we've talked about instances where we might not want to provide a label (but it's not interactive mode) but AFAIK this doesn't exist yet

Reply:
i got rid of this flag in the latest updates